test/e2e/framework/util.go

```diff
@@ -3482,7 +3482,17 @@ func GetSigner(provider string) (ssh.Signer, error) {
 		key = filepath.Join(keydir, keyfile)
 	}
 
-	return sshutil.MakePrivateKeySignerFromFile(key)
+	signer, err := sshutil.MakePrivateKeySignerFromFile(key)
+	if err != nil {
```

Comment:
Maybe something like this to simplify the nesting?
```go
if err == nil {
  return signer, nil
}

signer, err2 := sshutil.MakePrivateKeySignerFromAgent(keyfile)
if err2 == nil {
  return signer, nil
}

return nil, fmt.Errorf(".....")
```

Reply:
Done.